mw-webapp/src/logic/wayPage/GoalMetricsBlock.tsx

```diff
@@ -0,0 +1,253 @@
+import {useState} from "react";
+import {TrashIcon} from "@radix-ui/react-icons";
+import {Button} from "src/component/button/Button";
+import {Checkbox} from "src/component/checkbox/Сheckbox";
+import {EditableText} from "src/component/editableText/EditableText";
+import {HorizontalContainer} from "src/component/horizontalContainer/HorizontalContainer";
+import {Tooltip} from "src/component/tooltip/Tooltip";
+import {GoalMetricDAL} from "src/dataAccessLogic/GoalMetricDAL";
+import {useGlobalContext} from "src/GlobalContext";
+import {renderModalContent} from "src/logic/wayPage/reportsTable/WayColumns";
+import {Goal} from "src/model/businessModel/Goal";
+import {GoalMetric} from "src/model/businessModel/GoalMetric";
+import {Way} from "src/model/businessModel/Way";
+import {DateUtils} from "src/utils/DateUtils";
+import {v4 as uuidv4} from "uuid";
+import styles from "src/logic/wayPage/WayPage.module.scss";
+
+/**
+ * Singular metric for goal
+ */
+class SingleGoalMetric {
+
+  /**
+   * Goal metric uuid
+   */
+  public uuid: string;
+
+  /**
+   * Metric uuid
+   */
+  public metricUuid: string;
+
+  /**
+   * Metric description
+   */
+  public description: string;
+
+  /**
+   * Metric is done or not
+   */
+  public isDone: boolean;
+
+  /**
+   * Date when metric was done
+   */
+  public doneDate: Date;
+
+  constructor(params: SingleGoalMetric) {
+    this.uuid = params.uuid;
+    this.metricUuid = params.metricUuid;
+    this.description = params.description;
+    this.isDone = params.isDone;
+    this.doneDate = params.doneDate;
+  }
+
+}
+
+/**
+ * GoalMetricStatisticsBlock Props
+ */
+interface GoalMetricStatisticsBlockProps {
+
+  /**
+   * Is visible
+   * @default true
+   */
+  isVisible: boolean;
+
+  /**
+   * Way
+   */
+  way: Way;
+
+}
+
+/**
+ * Goal metrics block
+ */
+export const GoalMetricStatisticsBlock = (props: GoalMetricStatisticsBlockProps) => {
+  const [way, setWay] = useState<Way>(props.way);
+  const {user} = useGlobalContext();
+
+  const isOwner = !!user && user.uuid === way.owner.uuid;
```

Comment:
Why are you not passing it in props?

Reply:
Is it because of you want to have single point of truth?

Reply:
@tritonJS826 Yes, plus duplicate code in every block

Reply:
Deleted